main.py

```diff
@@ -0,0 +1,96 @@
+import argparse
+import os
+from pathlib import Path
+from typing import List
+import mediapipe as mp
+import cv2
+
+# IMAGE_EXTENSIONS = ['png', 'jpg']
+IMAGE_EXTENSIONS = ['mp4']
+DEFAULT_BASE_DIR = 'videos'
+POSE_INIT = mp.solutions.pose.Pose(
+    static_image_mode=False,  # change to False of was video stream
+    model_complexity=1,
+    smooth_landmarks=True,
+    min_detection_confidence=0.5,
+    min_tracking_confidence=0.5
+)
+
+DRAWING = mp.solutions.drawing_utils
+
+
+def detect_pose(frame,pose):
+    frame_rgb = cv2.cvtColor(frame,cv2.COLOR_BGR2RGB)
+    result = pose.process(frame_rgb)  # start detecting the keypoints
+    return result
+
+def draw_pose_on_frame(frame, detected_frame,  drawing):
+    if detected_frame.pose_landmarks:
+        DRAWING.draw_landmarks(
+            image=frame,
+            landmark_list=detected_frame.pose_landmarks,
+            connections=mp.solutions.pose.POSE_CONNECTIONS,
+            landmark_drawing_spec=DRAWING.DrawingSpec(
+                color=(250, 250, 250),
+                thickness=7,
+                circle_radius=4
+            ),
+            connection_drawing_spec=DRAWING.DrawingSpec(
+                color=(0, 0, 255),
+                thickness=11,
+                circle_radius=3,
+            )
+        )
+        print(detected_frame.pose_landmarks)
+
+def detect_human_pose_from_video(video_path: str):
+    pose = POSE_INIT
+    cap = cv2.VideoCapture(video_path)
+
+    while cap.isOpened():
+        ret,frame = cap.read()
+        if not ret:
+            break
+        # frame_to_RGB = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+        result = detect_pose(frame,POSE_INIT)
+        draw_pose_on_frame(frame,result,DRAWING)
+        cv2.imshow('Video After detecting the pose', frame)
+        if cv2.waitKey(1) & 0xFF == ord('q'):
+            break
+    cap.release()
+    cv2.destroyAllWindows()
+
+def main(argv=None) -> None:
```

Comment:
Separate code to small functions.  Read more: https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be